### PR TITLE
feat: Add support for Instagram Stories linkUrl/linkText

### PR DESCRIPTION
**Pull request by TimBroddin**
Related to #1590

# Overview
This PR adds support for the linkUrl/link Instagram Stories attributes. Instagram introduced new changes that rendered empty links for many. This PR allows you to set the link/text.

Kudos to @maxbb for the fix! 

| Before | After |
|-|-|
| Schermafbeelding 2024-10-25 om 09 54 27 | Schermafbeelding 2024-10-25 om 09 56 48 | 

Passing nothing (or an empty string for link/text) still renders the empty link.

# Test Plan
1. Apply this PR 
2. Make some code that shares something to Instagram Stories:

```ts
 await Share.shareSingle({
                    stickerImage: dataURI,
                    social: app.scheme,
                    appId: 'xxxx',
                    backgroundBottomColor: '#fefefe',
                    backgroundTopColor: '#906df4',
                    linkUrl: 'https://www.endgame.now/',
                    linkText: t`Download this app`,
                });
```

I was not able to test this on Android as I don't have a testing env with Instagram app set up.

### CODE REVIEW

**Comment by ferrannp** [comment]
I can confirm I see the empty link. Thanks for this!

**Comment by ferrannp** [comment]
Any way to make the link appear at the bottom and not at the middle? I guess no...? But doesn't hurt to ask 😅 . I was hoping to remove it too...

**Comment by TimBroddin** [comment]
> Any way to make the link appear at the bottom and not at the middle? I guess no...? But doesn't hurt to ask 😅 . I was hoping to remove it too...

I don't know. I hope Instagram documents these parameters soon :)

File: src/types.ts [hunk 1]
```diff
@@ -72,6 +72,8 @@ interface BaseSocialStoriesShareSingleOptions extends Omit<BaseShareSingleOption
   backgroundTopColor?: string;
   attributionURL?: string;
   backgroundVideo?: string;
+  linkUrl?: string;
+  linkText?: string;
```
Comment (MateusAndrade) [inline src/types.ts:76]: Can we add this to the docs?

Comment (TimBroddin) [reply src/types.ts:76]: I added it. Thanks for the review!

**Comment by mikehardy** [comment]
I'm running these locally since our CI is busted, I'll fix the conflict and re-push @TimBroddin no worries

**Comment by TimBroddin** [comment]
Nice, thanks a lot!

**Comment by mikehardy** [comment]
Couldn't push back to the branch but I'm trying to get as much done here in the little time I have available to donate to react-native-share so I just made a fresh PR with same commits rebased / conflict resolved and merged that. It's in!
Will run a release in a moment
Thanks again for the PR

**Comment by mikehardy** [comment]
@MateusAndrade hey Mateus 👋  I looked more closely and it looks like release will happen 100% automatically on merge but only for authorized folks, and I am not authorized in circle-ci 😄 - so my merges did not trigger a release and in fact I cannot trigger a release, I believe you'll have to do it.

Additionally it looks like CircleCI may have changed it's policy such that free+open-source repos get free linux build minutes, but no mac runner minutes, so perhaps that is the cause of iOS pipeline steps no longer running, causing things to get stuck in CI here?

**Pull request closed without merging**